### PR TITLE
Remove SHOUTY BANNERS

### LINKED ISSUE #306: STL: WE SHOULD REMOVE THE SHOUTY COMMENT BANNERS!
We currently follow a convention of introducing structs, classes, functions, etc. with "SHOUTY COMMENT BANNERS". A few examples:

https://github.com/microsoft/STL/blob/580e61a5f5536016f521cc1b2ca636eadc69bd30/stl/inc/vector#L410-L412

https://github.com/microsoft/STL/blob/580e61a5f5536016f521cc1b2ca636eadc69bd30/stl/inc/algorithm#L29-L41

https://github.com/microsoft/STL/blob/580e61a5f5536016f521cc1b2ca636eadc69bd30/stl/inc/type_traits#L536-L541

We've been thinking about changing this convention. It hasn't been applied rigorously, even for publicly-documented components (e.g. we have `// STRUCT TEMPLATE is_empty` but not `// VARIABLE TEMPLATE is_empty_v`). It might have helped file navigation many years ago, but IDE/editor support for Go To Definition and Find In Files (e.g. VSCode Ctrl+Shift+F) makes it easy to find where components are defined.

In theory, the banners could be most useful when publicly-documented components are preceded by internal support machinery, but in practice, we haven't always used them like that. For example, some but not all of the helpers for `is_permutation()` are introduced with `// FUNCTION TEMPLATE is_permutation`:

https://github.com/microsoft/STL/blob/580e61a5f5536016f521cc1b2ca636eadc69bd30/stl/inc/xutility#L4395-L4455

As these comments seem to provide minimal value today, are commonly overlooked by new contributors, and consume valuable lines, we should decide whether to keep or discard them.

### PR DESCRIPTION
**Pull request by SuperWig**
Continuation of #2072. I kept getting issues with the uppercase name of the branch so I renamed it. Didn't think it would close the PR  😕.

Fixes #306

### CODE REVIEW

File: stl/src/locale0.cpp
Comment (SuperWig) [inline stl/src/locale0.cpp]: This was accidentally deleted by me and pointed out by @AlexGuteniev

Is this not just a shouty banner in disguise?

File: stl/inc/charconv
Comment (SuperWig) [inline stl/inc/charconv:36]: Should "INTEGER TO STRING" part (and the others below) remain?

File: stl/inc/xutility
Comment (AlexGuteniev) [inline stl/inc/xutility:3362]: why did formatting change

File: stl/src/locale.cpp
Comment (AlexGuteniev) [inline stl/src/locale.cpp:4]: huh?

File: stl/inc/xutility
Comment (SuperWig) [reply stl/inc/xutility:3362]: Because I'm way too reliant on clang-format and forgot there's regions where that's off 🤦‍♂️. Actually just noticed this when removing the latest batch.

File: stl/src/locale.cpp
Comment (SuperWig) [reply stl/src/locale.cpp:4]: I simply missed one https://github.com/microsoft/STL/pull/2074#discussion_r674914034. We can have it both ways till a maintainer chimes in? 😅

**Comment by SuperWig** [comment]
That is almost all of them now. Will finish up tomorrow along with fixing my inevitable mistakes.

File: stl/inc/xmemory
Comment (CaseyCarter) [inline stl/inc/xmemory:902]: Ditto "there's useful information being lost here". I suggest `// See N4892 [unord.map.overview]/4`.

File: stl/inc/xutility
Comment (CaseyCarter) [inline stl/inc/xutility:1449]: Ditto; `// deduction guide utilities (N4892 [associative.general]/2)`.

**Review by CaseyCarter — CHANGES_REQUESTED** [review]
This is mostly looking good, just a few places where the comment was `GARBAGE name [other stuff..]` for which I think the `[other stuff...]` was just barely useful enough to keep.

**Comment by SuperWig** [comment]
I *think* that's it? Might have removed some useful ones in that last batch.

As far as I can tell the only remaining shouty comments are useful? The ones in [stl/src/xmath.hpp](https://github.com/SuperWig/STL/blob/a110d515cb133e68c83e09cd2c973ea3fb741481/stl/src/xmath.hpp) and the ones in [stl/inc/yvals.h](https://github.com/SuperWig/STL/blob/a110d515cb133e68c83e09cd2c973ea3fb741481/stl/inc/yvals.h) (found searching "PROPERTIES").

**Review by fsb4000 — APPROVED** [review]
Yes, 138/138 files viewed :)

**Comment by SuperWig** [comment]
Hopefully got them all now I have left some in `yvals(_core).h` as they might be helpful?

**Review by CaseyCarter — APPROVED** [review]

**Comment by CaseyCarter** [comment]
> Hopefully got them all now I have left some in `yvals(_core).h` as they might be helpful?

I prefer to err on the side of keeping things, so if you think they might be useful, let's keep'em. As large as this set of changes is, it's best to restrict it to changes that are as uncontroversial as possible. Anything questionable or that might need more attention / surgery we can do elsewhere which avoids blowing this up into something that takes multiple days to review.

Also - are you ready to make this a non-Draft PR?

**Comment by SuperWig** [comment]
> so if you think they might be useful

Only as they're blocks of macros and searching for the banner might be slightly easier? And yes, was just going to wait for an answer to that question.

**Comment by fsb4000** [comment]
add
```
Fixes #306
```
to your first comment.

File: stl/src/xmath.hpp
Comment (AlexGuteniev) [inline stl/src/xmath.hpp:10]: Doesn't look like useless. Maybe just decapitalize?

File: stl/src/xmath.hpp
Comment (AlexGuteniev) [inline stl/src/xmath.hpp:17]: Doesn't look like useless. Maybe just decapitalize?

File: stl/inc/ymath.h
Comment (AlexGuteniev) [inline stl/inc/ymath.h:20]: Not sure if useless. Maybe just decapitalize?

**Review by AlexGuteniev — CHANGES_REQUESTED** [review]
I think some macro comments should be preserved. They give non obvious context, since  macros lack type system, and these particular are not even verbose.

File: stl/src/xmath.hpp
Comment (SuperWig) [reply stl/src/xmath.hpp:17]: The ones in `yvals.h` could be de-capitalised?

https://github.com/microsoft/STL/blob/c2535c363805d8428790f3c2e2f2a139f5b90096/stl/inc/yvals.h#L34

https://github.com/microsoft/STL/blob/c2535c363805d8428790f3c2e2f2a139f5b90096/stl/inc/yvals.h#L310

https://github.com/microsoft/STL/blob/c2535c363805d8428790f3c2e2f2a139f5b90096/stl/inc/yvals.h#L316-L317

Comment (AlexGuteniev) [reply stl/src/xmath.hpp:17]: I less care about those, these macros are clearer without comments than those I pointed out

**Review by CaseyCarter — APPROVED** [review]

**Review by AlexGuteniev — APPROVED** [review]

File: stl/inc/variant
Comment (StephanTLavavej) [inline stl/inc/variant:938]: There's at least one shouty banner below that's not being removed:

https://github.com/microsoft/STL/blob/6a53cf2b809cbd3c10d5f35226e996ad5486ac06/stl/inc/variant#L1256

File: stl/inc/variant
Comment (StephanTLavavej) [inline stl/inc/variant:378]: These stable name citations are being dropped (also below). @CaseyCarter should they be preserved by themselves?

Here is the full list of stable names being dropped:

```cpp
// STRUCT nullopt_t [optional.nullopt]
// CLASS bad_optional_access [optional.bad_optional_access]
// CLASS TEMPLATE optional [optional.object]
// COMPARISONS WITH nullopt [optional.nullops]
// COMPARISONS WITH T [optional.comp_with_t]
// FUNCTION TEMPLATE swap [optional.specalg]
// FUNCTION TEMPLATE make_optional [optional.specalg]
// STRUCT TEMPLATE SPECIALIZATION hash [optional.hash]
// VARIANT OF VALUE TYPES [variant.variant]
// VARIANT HELPER CLASSES [variant.helper]
// CLASS bad_variant_access [variant.bad.access]
// CLASS monostate [variant.monostate]
```

**Review by StephanTLavavej — COMMENTED** [review]
Thanks - I'll go ahead and push minor changes.

File: stl/src/xmath.hpp
Comment (StephanTLavavej) [reply stl/src/xmath.hpp:17]: After looking at them, I dropped CURRENT DLL NAMES, MULTITHREAD PROPERTIES, and LOCK MACROS as being obvious/unnecessary, but lowercased INTEGER PROPERTIES because their names were kind of hard to understand otherwise.

**Comment by StephanTLavavej** [comment]
Ok, I've pushed a merge with main (trivial to resolve - recent PRs added/altered code around stable name removals) and the following:

* Restore stable names for optional and variant.
* In variant/optional, drop more shouty banners but keep stable names.
* Remove many more shouty banners.
* Rework shouty banners to preserve info.

In optional and variant I'd be happy to remove those stable name citations entirely, but as some were already present with lowercase comments like:

https://github.com/microsoft/STL/blob/6a53cf2b809cbd3c10d5f35226e996ad5486ac06/stl/inc/variant#L1089

I felt it was more consistent to keep them for now.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Removed the optional/variant stable name comments (including the ones of the form `// assignment [optional.object.assign]`) after @CaseyCarter confirmed they weren't desired.

**Review by StephanTLavavej — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
THANKS FOR CLEANING UP THESE COMMENTS! :joy_cat:
===

**Pull request merged**
